actions/images

```diff
@@ -0,0 +1,38 @@
+#!/usr/local/sbin/charm-env python3
+
+import sys
+from subprocess import check_output, CalledProcessError, STDOUT
+
+from charmhelpers.core import hookenv
+from charms import layer
+from charms.reactive import is_flag_set
+
+layer.import_layer_libs()
+
+# start reactive
+hookenv._run_atstart()
+
+
+def fail(msg):
+    hookenv.action_fail(msg)
+    sys.exit()
```

Comment:
Technically this should be `sys.exit(msg)` or `sys.exit(1)`. As is, juju will report that this action completed, when it actually failed.

Reply:
When using `sys.exit($non-zero)`, juju shows a generic return code error, and the overall action rc is still 0:
```bash
## with sys.exit(1), unhelpful message
$ juju run-action --wait docker-registry/0 images options=-badtim
unit-docker-registry-0:
  id: 964b7a31-3080-4853-82c0-a002c99f3cf0
  message: exit status 1
  status: failed
  timing:
    completed: 2019-04-26 21:14:27 +0000 UTC
    enqueued: 2019-04-26 21:14:26 +0000 UTC
    started: 2019-04-26 21:14:26 +0000 UTC
  unit: docker-registry/0
$ echo $?
0

## with sys.exit('this is fine'), unhelpful message; supplied msg goes to debug log
$ juju run-action --wait docker-registry/0 images options=-badtim
unit-docker-registry-0:
  id: 0f743990-50a7-4f95-84fd-9729b977ec90
  message: exit status 1
  status: failed
  timing:
    completed: 2019-04-26 21:13:31 +0000 UTC
    enqueued: 2019-04-26 21:13:29 +0000 UTC
    started: 2019-04-26 21:13:30 +0000 UTC
  unit: docker-registry/0
$ echo $?
0
```

Whereas `sys.exit($zero)` allows our helpful `action_fail` message to come through:
```
$ juju run-action --wait docker-registry/0 images options=-badtim
unit-docker-registry-0:
  id: 811696ab-75c8-4dcf-8de7-e291dbfa625f
  message: |
    "docker images -badtim" failed with:
    unknown shorthand flag: 'b' in -badtim
    See 'docker images --help'.
  status: failed
  timing:
    completed: 2019-04-26 21:17:20 +0000 UTC
    enqueued: 2019-04-26 21:17:16 +0000 UTC
    started: 2019-04-26 21:17:19 +0000 UTC
  unit: docker-registry/0
$ echo $?
0
```

Reply:
Huh, TIL.